dartagnan/src/test/java/com/dat3m/dartagnan/DataRaceTest.java

```diff
@@ -0,0 +1,131 @@
+package com.dat3m.dartagnan;
+
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.utils.Result;
+import com.dat3m.dartagnan.utils.Settings;
+import com.dat3m.dartagnan.utils.rules.CSVLogger;
+import com.dat3m.dartagnan.utils.rules.Provider;
+import com.dat3m.dartagnan.utils.rules.Providers;
+import com.dat3m.dartagnan.utils.rules.RequestShutdownOnError;
+import com.dat3m.dartagnan.verification.VerificationTask;
+import com.dat3m.dartagnan.wmm.Wmm;
+import com.dat3m.dartagnan.wmm.utils.Arch;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.sosy_lab.common.ShutdownManager;
+import org.sosy_lab.java_smt.api.ProverEnvironment;
+import org.sosy_lab.java_smt.api.SolverContext;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static com.dat3m.dartagnan.analysis.DataRaces.checkForRaces;
+import static com.dat3m.dartagnan.utils.ResourceHelper.TEST_RESOURCE_PATH;
+import static com.dat3m.dartagnan.utils.Result.FAIL;
+import static com.dat3m.dartagnan.utils.Result.PASS;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class DataRaceTest {
+
+    protected String name;
+    protected int bound;
+
+    public DataRaceTest(String name, int bound) {
+        this.name = name;
+        this.bound = bound;
+    }
+
+    // =================== Modifiable behavior ====================
+
+    protected Provider<String> getProgramPathProvider() {
+        return Provider.fromSupplier(() -> TEST_RESOURCE_PATH + "boogie/data-races/" + name + "-O0.bpl");
+    }
+
+    protected long getTimeout() { return 180000; }
+
+    protected Provider<Settings> getSettingsProvider() {
+        return Provider.fromSupplier(() -> new Settings(bound, 0));
+    }
+
+    protected Provider<Wmm> getWmmProvider() {
+        return GlobalSettings.ATOMIC_AS_LOCK ?
+                Providers.createWmmFromName(() -> "svcomp-locks") :
+                Providers.createWmmFromName(() -> "svcomp");
+    }
```

Comment:
These methods can get "inlined" unless you intend to create subclasses of this test class to override them (that is the reason by these methods exist in the abstract base classes). But even without inheritence in mind, it is not bad to have these methods. So it is up to you if you want to keep them.